src/code/z_kankyo.c

```diff
@@ -190,27 +174,40 @@ SkyboxFile gSkyboxFiles[] = {
     },
 };
 
-u8 D_8011FDCC = 0;
-u8 D_8011FDD0 = 0;
-f32 D_8011FDD4 = 0.0f;
+u8 sSandstormColorIndex = 0;
+u8 sNextSandstormColorIndex = 0;
+f32 sSandstormLerpScale = 0.0f;
 
 u8 gCustomLensFlareOn;
 Vec3f gCustomLensFlarePos;
 s16 gLensFlareUnused;
 s16 gLensFlareScale;
 f32 gLensFlareColorIntensity;
 s16 gLensFlareScreenFillAlpha;
+
+typedef struct {
+    /* 0x00 */ u8 state;
+    /* 0x04 */ Vec3f offset;
+    /* 0x10 */ Vec3f pos;
+    /* 0x1C */ s8 pitch;
+    /* 0x1D */ s8 roll;
+    /* 0x1E */ u8 textureIndex;
+    /* 0x1F */ u8 delayTimer;
+} LightningBolt; // size = 0x20
+
 LightningBolt sLightningBolts[3];
 LightningStrike gLightningStrike;
 s16 sLightningFlashAlpha;
-s16 D_8015FD7E;
-s16 D_8015FD80;
+
+s16 sLensFlareDepthTestX;
+s16 sLensFlareDepthTestY;
+
 LightNode* sNGameOverLightNode;
 LightInfo sNGameOverLightInfo;
 LightNode* sSGameOverLightNode;
 LightInfo sSGameOverLightInfo;
 u8 sGameOverLightsIntensity;
-u16 D_8015FDB0;
+u16 sSandstormScroll;
 
 s32 func_8006F0A0(s32 a0) {
     s32 ret = ((a0 >> 4 & 0x7FF) << D_8011FAF0[a0 >> 15 & 7].unk0) + D_8011FAF0[a0 >> 15 & 7].unk1;
```

Comment:
Not so much a suggestion as much as me leaving my rambling thoughts here for anyone else interested in figuring out exactly what's going on here.
At a distance, this function is responsible for converting a z-buffer depth value (more specifically, a depth value `* 4` or `<< 2`) to a screen depth value (as in, `z` in screen space)
The contents of the z-buffer is an interesting format. It's 16 bits in 2 parts, `z` and `dz`. `z` is a 3.11 fixed point number so takes up 14 bits, the remaining 2 bits are for `dz`. The maximum depth value is encapsulated in the constant `G_MAXFBZ` (`0x3FFF`). On the other hand, the "screen depth" value is a floating point number with a different scale.
The `unk1` member of `D_8011FAF0` follows a sequence where the difference between each value is half the previous difference.

Reply:
This from where `func_8006F0A0` is called
```c
wZ = (s32)((multDest.z * wDest) * 16352.0f) + 16352;
```
is straight out of the manual (12.7.6 Depth Source)
>     `screenZ=32*((z/w)*Viewport.vscale[2]+Viewport.vtrans[2])`
> Note: `Viewport.vscale` and `Viewport.vtrans[2]` are usually both `G_MAXZ/2 = 0x1ff`, which makes the formula: `screenZ=(z/w)*0x3fe0 + 0x3fe0`. Since `(z/w)` ranges from -1.0 to +1.0 the result will range from 0x0 to 0x7fc0. 

In the next line `zBuf = gZBuffer[(s32)((wY * -120.0f) + 120.0f)][(s32)((wX * 160.0f) + 160.0f)] * 4` I guess the `* 4` (`<< 2`) accounts for the actual zbuffer value being 18 bits (two 9-bits bytes), this is backed up by this value then being passed to `func_8006F0A0` which seems to consider it as 18 bits

In `func_8006F0A0`
the dz value (`a0 & 0xF`) seems to be ditched
`a0 >> 4 & 0x7FF` : 11 lower bits of the z value
`a0 >> 15 & 7` : 3 higher bits of the z value

for `unk1` in `D_8011FAF0` looking at them in binary is what I think reveals structure but I have no idea what the function is doing with it, doesn't make sense to me